service/create/encrypt.go

```diff
@@ -0,0 +1,161 @@
+package create
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/aws/aws-sdk-go/aws"
+	awssession "github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/kms"
+	microerror "github.com/giantswarm/microkit/error"
+)
+
+// PEM encoded TLS assets.
+type rawTLSAssets struct {
+	APIServerCACert    []byte
+	APIServerKey       []byte
+	APIServerCert      []byte
+	CalicoClientCACert []byte
+	CalicoClientKey    []byte
+	CalicoClientCert   []byte
+	EtcdServerCACert   []byte
+	EtcdServerKey      []byte
+	EtcdServerCert     []byte
+}
+
+// Encrypted PEM encoded TLS assets
+type encryptedTLSAssets struct {
+	APIServerCACert    []byte
+	APIServerKey       []byte
+	APIServerCert      []byte
+	CalicoClientCACert []byte
+	CalicoClientKey    []byte
+	CalicoClientCert   []byte
+	EtcdServerCACert   []byte
+	EtcdServerKey      []byte
+	EtcdServerCert     []byte
+}
+
+// PEM -> encrypted -> gzip -> base64 encoded TLS assets.
+type CompactTLSAssets struct {
+	APIServerCACert    string
+	APIServerKey       string
+	APIServerCert      string
+	CalicoClientCACert string
+	CalicoClientKey    string
+	CalicoClientCert   string
+	EtcdServerCACert   string
+	EtcdServerKey      string
+	EtcdServerCert     string
+}
+
+func readRawTLSAssets(tlsAssetsDir string) (*rawTLSAssets, error) {
```

Comment:
I think that this should be separated as well. Like, the KVM operator wants to do the same thing, right? It wants to provide certificates to the cloud configs. Here you mix that up with some AWS specifics in `encrcypt` below. IMO we should not mix this, because the KVM operator is not concerned about AWS at all.